src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -16,21 +16,25 @@
     /**
      * Executes the command and returns the result message.
      *
-     * @param model {@code Model} which the command should operate on.
-     * @return feedback message of the operation result for display
+     * @param model {@code Model} which contains the model which the command operates on
      * @throws CommandException If an error occurs during command execution.
      */
-    public abstract CommandResult execute(Model model) throws CommandException;
-
-    protected void requireDeliveryModel(Model model) throws CommandException {
-        if (!(model instanceof DeliveryModel)) {
-            throw new CommandException(MESSAGE_INCORRECT_MODEL);
+    public CommandResult execute(Model model) throws CommandException {
+        if (model instanceof InventoryModel) {
+            return execute((InventoryModel) model, new DeliveryModelManager());
+        } else {
+            return execute(new InventoryModelManager(), (DeliveryModelManager) model);
```

Comment:
@halcon-blanco as discussed in telegram, we should probably create a wrapper class so we can make use of polymorphism instead of using instanceof

Reply:
But for now let's implement the rest on top of this first

Reply:
@halcon-blanco i have a suggestion. We should make use of DeliveryCommand and ItemCommand abstract class to have a abstract method execute that takes in its respective model. This will adhere to OpenClosed principle better